homeassistant/components/sensor/filesize.py

```diff
@@ -0,0 +1,105 @@
+"""
+Sensor for monitoring the size of a file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filesize/
+"""
+import datetime
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_FILE_PATHS = 'file_paths'
+ICON = 'mdi:file'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FILE_PATHS):
+        vol.All(cv.ensure_list, [cv.isfile]),
+})
+
+
+def get_file_size(path):
+    """Return the size of the file in bytes."""
+    statinfo = os.stat(path)
+    file_size_bytes = statinfo.st_size
+    return file_size_bytes
+
+
+def get_last_updated(path):
+    """Return the time the file was last modified."""
+    statinfo = os.stat(path)
+    last_updated = datetime.datetime.fromtimestamp(statinfo.st_mtime)
+    last_updated = last_updated.isoformat(' ')
+    return last_updated
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the file size sensor."""
+    sensors = []
+    for path in config.get(CONF_FILE_PATHS):
+        if not hass.config.is_allowed_path(path):
+            _LOGGER.error(
+                "Filepath %s is not valid or allowed", path)
```

Comment:
Make sure to make a note in the docs how to change allowed paths

Reply:
Done. For whitelist_external_dirs I've linked https://home-assistant.io/docs/configuration/basic/ but perhaps theres a better page to point to?